src/control/zoom/ZoomControl.cpp

```diff
@@ -157,7 +184,9 @@ void ZoomControl::addZoomListener(ZoomListener* l) { this->listener.emplace_back
 void ZoomControl::initZoomHandler(GtkWidget* window, GtkWidget* widget, XournalView* v, Control* c) {
     this->control = c;
     this->view = v;
+    gtk_widget_add_events(widget, GDK_TOUCHPAD_GESTURE_MASK);
     g_signal_connect(widget, "scroll-event", G_CALLBACK(onScrolledwindowMainScrollEvent), this);
+    g_signal_connect(widget, "event", G_CALLBACK(onTouchpadPinchEvent), this);
     g_signal_connect(window, "configure-event", G_CALLBACK(onWindowSizeChangedEvent), this);
```

Comment:
Not sure whether I'm just not able to find it or it really isn't there/is that bad, but I couldn't really find any documentation on this GTK/GDK event stuff. I'm not really sure about what exactly the "event" parameter does, but changing it to something like "touchpadpinch-event" didn't work. Also, I don't know whether this is the right place to do the gtk_widget_add_events call.

Reply:
Alright, found the signals it can connect to (https://developer.gnome.org/gtk3/stable/GtkWidget.html#GtkWidget.signals) - as there doesn't seem to be anything specific to touchpad gestures I think we have to keep this general event handler.
Question to you then is: should I rename onTouchpadPinchEvent as it actually is a general event handler or keep it this way as it effectively only handles pinch events?

Reply:
Please keep the name as it is. If we ever need to handle more events it should be obvious that this handler is only for the pinch events.